interfaces/prompting/requestprompts/requestprompts.go

```diff
@@ -281,10 +350,41 @@ func (pdb *PromptDB) AddOrMerge(metadata *prompting.Metadata, path string, reque
 
 	userEntry, ok := pdb.perUser[metadata.User]
 	if !ok {
-		pdb.perUser[metadata.User] = &userPromptDB{
+		// New user entry, so create it and set up the expiration timer
+		userEntry = &userPromptDB{
 			ids: make(map[prompting.IDType]int),
 		}
-		userEntry = pdb.perUser[metadata.User]
+		userEntry.expirationTimer = time.AfterFunc(initialTimeout, func() {
```

Comment:
the function should probably just be a call to a method on pdb with the bulk of the logic taking a userEntry